app/models/discovery_attribute_set.rb

```diff
@@ -0,0 +1,8 @@
+class DiscoveryAttributeSet < ActiveRecord::Base
+  belongs_to :host, :class_name => "Host::Discovered", :foreign_key => :host_id
+  attr_accessible :cpu_count, :disk_count, :disks_size, :memory
+
+  validates :cpu_count, :presence => true, :numericality => {:greater_than_or_equal_to => 0}
+  validates :memory, :presence => true,    :numericality => {:greater_than_or_equal_to => 0}
+  validates :host, :presence => true
+end
```

Comment:
I was not following your discussion about introducing new columns and I take it it is not possible to join against facts table. With Facter 2.0 memory (and cpu_count) are reported as numbers instead of strings. Just thinking loud.